### PR TITLE
Add more targets to prune whitelist

### PR DESCRIPTION
**Pull request by RyanBrushett**
**What are you trying to accomplish with this PR?**

Quick win: resolve https://github.com/Shopify/kubernetes-deploy/issues/271
Basically, expand the prune whitelist to include ServiceAccount, PodTemplate, and ReplicaSet.

**How is this accomplished?**

I just added them to the whitelist. Seems consistent.

**What could go wrong?**

Could prune stuff it shouldn't or miss things it should prune, I suppose. Does this cover enough?

### CODE REVIEW

**Comment by KnVerey** [comment]
> Does this cover enough?

Can you explain more how you came up with the list? Which namespaced resources, if any, are you excluding from it?

**Comment by RyanBrushett** [comment]
@KnVerey Your issue here: https://github.com/Shopify/kubernetes-deploy/issues/271
Didn't think much harder about it than that, honestly. Should that list be expanded?

**Comment by KnVerey** [comment]
Ah, I see. Yeah, that issue was opened more than a year ago. I'm thinking for the purposes of this PR, what we need is a quick audit of the resources we support vs. what is on the whitelist. I think all namespaced resources we support should be on it, but maybe I'm forgetting something.

**Comment by RyanBrushett** [comment]
@KnVerey Good shout, I'll do some digging and see what I can come up with 👍

**Comment by RyanBrushett** [comment]
@KnVerey I went through our `kubernetes_resource` directory to see what k8s resources we support (aside from custom stuff) and what wasn't in our prune whitelist.
I added `role` and `rolebinding` which are in our `kubernetes_resource` directory, are namespaced, but aren't currently in our prune whitelist. Does that make sense?

File: lib/kubernetes-deploy/deploy_task.rb [hunk 1]
```diff
@@ -58,7 +58,6 @@ class DeployTask
     # core/v1/Endpoints -- managed by services
     # core/v1/PersistentVolumeClaim -- would delete data
```
Comment (KnVerey) [inline lib/kubernetes-deploy/deploy_task.rb]: I wonder if we should re-assess this choice as well. It is true that deleting a PVC can result in data loss (depending on the persistentVolumeReclaimPolicy on the underlying PV), but failing to prune them can result in continuing costs for undesired, unused storage.

Comment (RyanBrushett) [reply lib/kubernetes-deploy/deploy_task.rb]: Oooo good call! I'll do some more digging.

**Comment by RyanBrushett** [comment]
I'm leaning towards pruning persistent volume claims.
See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reclaiming

TL;DR One can set a reclaim policy on the volume to either retain or delete. If people want to be really careful, a retain reclaim policy will save them from losing data upon pruning a PVC.
Dynamically-provisioned volumes inherit the reclaim policy from their [storage class](https://kubernetes.io/docs/concepts/storage/storage-classes/) so, again, there is an opportunity to protect data.

Thoughts?

**Comment by RyanBrushett** [comment]
After some digging, it seems like it’d be a good idea to split this apart. Pruning PVCs could easily lead to data loss and I think that’s going to warrant some additional legwork.
See https://github.com/Shopify/kubernetes-deploy/issues/519 
Meanwhile, I think that’ll unblock this PR :+1: 

Thoughts anyone?
